core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/PhysicalOp.scala

```diff
@@ -0,0 +1,468 @@
+package edu.uci.ics.amber.engine.architecture.deploysemantics
+
+import akka.actor.Deploy
+import akka.remote.RemoteScope
+import edu.uci.ics.amber.engine.architecture.common.{AkkaActorRefMappingService, AkkaActorService}
+import edu.uci.ics.amber.engine.architecture.controller.{ControllerConfig, OperatorExecution}
+import edu.uci.ics.amber.engine.architecture.deploysemantics.layer.{
+  OpExecInitInfo,
+  OpExecInitInfoWithCode,
+  OpExecInitInfoWithFunc
+}
+import edu.uci.ics.amber.engine.architecture.deploysemantics.locationpreference.{
+  AddressInfo,
+  LocationPreference,
+  PreferController,
+  RoundRobinPreference
+}
+import edu.uci.ics.amber.engine.architecture.pythonworker.PythonWorkflowWorker
+import edu.uci.ics.amber.engine.architecture.worker.WorkflowWorker
+import edu.uci.ics.amber.engine.architecture.worker.WorkflowWorker.WorkflowWorkerConfig
+import edu.uci.ics.amber.engine.common.virtualidentity.{
+  ActorVirtualIdentity,
+  OperatorIdentity,
+  PhysicalLinkIdentity,
+  PhysicalOpIdentity
+}
+import edu.uci.ics.amber.engine.common.{AmberConfig, VirtualIdentityUtils}
+import edu.uci.ics.texera.workflow.common.metadata.{InputPort, OperatorInfo, OutputPort}
+import edu.uci.ics.texera.workflow.common.tuple.schema.{OperatorSchemaInfo, Schema}
+import edu.uci.ics.texera.workflow.common.workflow.{HashPartition, PartitionInfo, SinglePartition}
+import edu.uci.ics.texera.workflow.operators.hashJoin.HashJoinOpExec
+import org.jgrapht.graph.{DefaultEdge, DirectedAcyclicGraph}
+import org.jgrapht.traverse.TopologicalOrderIterator
+
+import scala.collection.mutable.ArrayBuffer
+
+object PhysicalOp {
+
+  /** all source operators should use sourcePhysicalOp to give the following configs:
+    *  1) it initializes at the controller jvm.
+    *  2) it only has 1 worker actor.
+    *  3) it has no input ports.
+    */
+  def sourcePhysicalOperator(
```

Comment:
Is there a reason for this to be named differently from `sourcePhysicalOp`? Previously the two had the same name.

Reply:
changed to `sourcePhysicalOp`